### DISCUSSION TITLE
Tuya TZE204

### DISCUSSION THREAD

**Discussion by timewolf** [Koenkk/zigbee2mqtt#16926] [Devices]
Hi,

i've one of this

https://it.aliexpress.com/item/1005004881907096.html

2P

I've found some .js template to use with zigbee2mqtt but not all seems ok

here my current external_converter

`const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require("zigbee-herdsman-converters/lib/tuya");

const definition = {

        fingerprint: tuya.fingerprint('TS0601', ['_TZE204_davzgqq0']),
        model: 'TS0601_din_2',
        vendor: 'TuYa',
        description: 'Zigbee DIN energy meter',
        fromZigbee: [tuya.fz.datapoints],
        toZigbee: [tuya.tz.datapoints],
        configure: tuya.configureMagicPacket,
        whiteLabel: [{vendor: 'XOCA', model: 'DAC2161C'}],
        exposes: [tuya.exposes.switch(), e.energy(), e.power(), e.voltage(), e.current(),
            exposes.enum('fault', ea.STATE, ['clear', 'over_current_threshold', 'over_power_threshold',
                'over_voltage threshold', 'wrong_frequency_threshold']).withDescription('Fault status of the device (clear = nothing)'),
            exposes.enum('threshold_1', ea.STATE, ['not_set', 'over_current_threshold', 'over_voltage_threshold'])
                .withDescription('State of threshold_1'),
            exposes.binary('threshold_1_protection', ea.STATE, 'ON', 'OFF')
                .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
            exposes.numeric('threshold_1_value', ea.STATE)
                .withDescription('Can be in Volt or Ampere depending on threshold setting. Setup the value on the device'),
            exposes.enum('threshold_2', ea.STATE, ['not_set', 'over_current_threshold', 'over_voltage_threshold'])
                .withDescription('State of threshold_2'),
            exposes.binary('threshold_2_protection', ea.STATE, 'ON', 'OFF')
                .withDescription('OFF - alarm only, ON - relay will be off when threshold reached'),
            exposes.numeric('threshold_2_value', ea.STATE)
                .withDescription('Setup value on the device'),
            exposes.binary('clear_fault', ea.STATE_SET, 'ON', 'OFF')
                .withDescription('Turn ON to clear last the fault'),
            exposes.text('meter_id', ea.STATE).withDescription('Meter ID (ID of device)'),
        ],
        meta: {
            tuyaDatapoints: [
                [1, 'energy', tuya.valueConverter.divideBy100],
                [3, null, null], // Monthly, but sends data only after request
                [4, null, null], // Dayly, but sends data only after request
                [6, null, tuya.valueConverter.phaseVariant2], // voltage and current
                [10, 'fault', tuya.valueConverterBasic.lookup({'clear': 0, 'over_current_threshold': 1,
                    'over_power_threshold': 2, 'over_voltage_threshold': 4, 'wrong_frequency_threshold': 8})],
                [11, null, null], // Frozen - strange function, in native app - nothing is clear
                [16, 'state', tuya.valueConverter.onOff],
                [17, null, tuya.valueConverter.threshold], // It's settable, but can't write converter
                [18, 'meter_id', tuya.valueConverter.raw],
                [20, 'clear_fault', tuya.valueConverter.onOff], // Clear fault
                [21, null, null], // Forward Energy T1 - don't know what this
                [22, null, null], // Forward Energy T2 - don't know what this
                [23, null, null], // Forward Energy T3 - don't know what this
                [24, null, null], // Forward Energy T4 - don't know what this
            ],
        },

};

module.exports = definition;

`

in log info i'm flooded by

`Info 2023-03-05 12:47:35MQTT publish: topic 'zigbee2mqtt/Prese Allarme', payload '{"clear_fault":null,"current":0.09,"device":{"applicationVersion":70,"dateCode":"","friendlyName":"Prese Allarme","hardwareVersion":1,"ieeeAddr":"0x84b4dbfffec09000","manufacturerID":4098,"manufacturerName":"_TZE204_davzgqq0","model":"TS0601_din_2","networkAddress":38706,"powerSource":"Mains (single phase)","stackVersion":0,"type":"Router","zclVersion":3},"energy":0.86,"fault":null,"linkquality":168,"meter_id":{"0":1,"1":1,"10":0,"11":180,"2":0,"3":38,"4":3,"5":1,"6":1,"7":9,"8":4,"9":1},"power":10,"state":"ON","threshold1_value":30,"threshold2_value":0,"threshold_1_protection":"ON","threshold_1_value":30,"threshold_2_value":0,"voltage":235.6}'
Info 2023-03-05 12:47:35MQTT publish: topic 'zigbee2mqtt/Prese Allarme', payload '{"clear_fault":null,"current":0.09,"device":{"applicationVersion":70,"dateCode":"","friendlyName":"Prese Allarme","hardwareVersion":1,"ieeeAddr":"0x84b4dbfffec09000","manufacturerID":4098,"manufacturerName":"_TZE204_davzgqq0","model":"TS0601_din_2","networkAddress":38706,"powerSource":"Mains (single phase)","stackVersion":0,"type":"Router","zclVersion":3},"energy":0.86,"fault":null,"linkquality":168,"meter_id":`

any help? :)

**Comment by tekkva** [comment]
Hi. Thank you. I tried it on one of this after changing the manufacturerName to _TZE204_wbhaespm.
https://www.amazon.se/gp/product/B0C4N9DTDX/ref=ppx_yo_dt_b_asin_image_o01_s00?ie=UTF8&psc=1 
It works fine but just used it for on and off so far but it is enough for me right now.
Gone do some more testing off coursers and hope I can help some to.
Thanks again.

**Comment by fred-c1** [comment]
About:

[3, null, null], // Monthly, but sends data only after request
[4, null, null], // Dayly, but sends data only after request

Any idea how to get DataPoints 3 and 4 working?

If needed, the "spec" should be:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/42831538/738fff6f-9017-48b3-923c-6cea8410420e)
